submissions/Chikus/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,69 @@
+const Net = require('net');
+var host = '';
+var lower_port = 0;
+var higher_port = 65535;
+var port_list = [];
+var timeout = 300;
+
+function check(host, port, callback) {
+ var socket = Net.createConnection(port, host);
+ var timer  = setTimeout(function () {
+  socket.destroy();
+  callback(false);
+}, timeout);
+ socket.once('connect', function () {
+  clearTimeout(timer);
+  socket.destroy();
+  process.stdout.write('.');
+  port_list.push(port);
+  callback(true);
+ });
+ socket.on('error', function () {
+  clearTimeout(timer);
+  callback(false);
+ });
+}
+
+function run_ports(){
+  check(host,lower_port, function next_check(result) {
+    if (lower_port == higher_port) {
+      if (port_list.length) {
+          process.stdout.write('\n'+port_list.join()+' ports are opened \n');
+      }
+      else {
+        process.stdout.write('No ports were found \n');
+      }
+    }
+    else {
+      check(host,++lower_port,next_check);
+    }
+  })
+};
+
+switch (process.argv[2]) {
+  case '--ports':
+    var ports = [2];
+    ports = process.argv[3].split('-');
+    lower_port = ports[0];
+    higher_port = ports[1];
```

Comment:
```js
const [lower_port, higher_port] = process.argv[3].split('-');
```

Also check for errors (string with no '-')

Reply:
Done!